File: aoBlogs2/Models/EmailModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Models
    Public Class EmailModel
        Inherits DbModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "Email"      '<------ set content name
        Public Const contentTableName As String = "ccEmail"   '<------ set to tablename for the primary content (used for cache names)
        Private Shadows Const contentDataSource As String = "default"             '<------ set to datasource if not default
        '
        '====================================================================================================
        ' -- instance properties

        Public Property AddLinkEID As Boolean
        Public Property AllowSpamFooter As Boolean
        Public Property BlockSiteStyles As Boolean
        Public Property ConditionExpireDate As Date
        Public Property ConditionID As Integer
        Public Property ConditionPeriod As Integer
        Public Property CopyFilename As String
        Public Property EmailTemplateID As Integer
        Public Property EmailWizardID As Integer
        Public Property FromAddress As String
        Public Property InlineStyles As String
        Public Property LastSendTestDate As Date
        Public Property ScheduleDate As Date
        Public Property Sent As Boolean
        Public Property StylesFilename As String
        Public Property Subject As String
        Public Property Submitted As Boolean
        Public Property TestMemberID As Integer
        Public Property ToAll As Boolean


    End Class
End Namespace

File: source/aoBlogs2/Models/Db/EmailModel.cs

//using Contensive.Models.Db;
//using System;

//namespace Contensive.Addons.Blog.Models {
//    public class EmailModel : Contensive.Models.Db.DbBaseModel {
//        // 
//        // ====================================================================================================
//        public static DbBaseTableMetadataModel tableMetadata { get; private set; } = new DbBaseTableMetadataModel("Email", "ccEmail", "default", false);
//        // -- const
//        public const string contentName = "Email";      // <------ set content name
//        public const string contentTableName = "ccEmail";   // <------ set to tablename for the primary content (used for cache names)
//        private new const string contentDataSource = "default";             // <------ set to datasource if not default
//        // 
//        // ====================================================================================================
//        // -- instance properties

//        public bool AddLinkEID { get; set; }
//        public bool AllowSpamFooter { get; set; }
//        public bool BlockSiteStyles { get; set; }
//        public DateTime ConditionExpireDate { get; set; }
//        public int ConditionID { get; set; }
//        public int ConditionPeriod { get; set; }
//        public string CopyFilename { get; set; }
//        public int EmailTemplateID { get; set; }
//        public int EmailWizardID { get; set; }
//        public string FromAddress { get; set; }
//        public string InlineStyles { get; set; }
//        public DateTime LastSendTestDate { get; set; }
//        public DateTime ScheduleDate { get; set; }
//        public bool Sent { get; set; }
//        public string StylesFilename { get; set; }
//        public string Subject { get; set; }
//        public bool Submitted { get; set; }
//        public int TestMemberID { get; set; }
//        public bool ToAll { get; set; }


//    }
//}